devp2p/discovery.py

```diff
@@ -62,7 +66,7 @@ def __init__(self, ip, udp_port, tcp_port=0, from_binary=False):
             # We only want v4 or v6 addresses
             # see https://docs.python.org/2/library/socket.html#socket.getaddrinfo
             ips = [
-                unicode(ai[4][0])
+                str(ai[4][0]) if PY3 else unicode(ai[4][0])
```

Comment:
I didn't remember why this was required but now checking again, it seems that `ipaddress.ip_address` only accepts `unicode`!

https://github.com/phihag/ipaddress/blob/master/ipaddress.py#L159